### PR TITLE
Doc: add CONTRIBUTING.md (derived from Bootstrap)

### PR DESCRIPTION
**Pull request by andythenorth**
This is enough for a v1.

* 3 or so different people have reviewed and contributed
* I've tested all links, they work for me 
* I didn't find any huge typos

### CODE REVIEW

**Comment by TrueBrain** [comment]
I really really like this :) Well done!

**Review by TrueBrain — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+Guidelines for bug reports:
+
+0. Please don't report issues with games where you changed NewGRFs.
+
+1. Please don't report issues with modified versions of OpenTTD (patchpacks and similar).
+
+2. **Use the GitHub issue search** --- check if the issue has already been
+reported.
+
+3. **Check if the issue has been fixed** --- try to reproduce it using the latest `nightly` build of OpenTTD, available from https://www.openttd.org/en/
```
Comment (frosch123) [inline CONTRIBUTING.md]: Remove the "en/"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+contain your feature, change, or fix:
+
+```bash
+git checkout upstream/master -b <topic-branch-name>
+```
+
+4. Commit your changes in logical chunks. Please adhere to these [git commit message guidelines](https://wiki.openttd.org/Commit_style#Commit_message) or your code is unlikely to be merged into the main project.
+Use Git's [interactive rebase](https://help.github.com/articles/interactive-rebase) feature to tidy up your commits before making them public.
+
+5. Locally merge (or rebase) the upstream development branch into your topic branch:
```
Comment (frosch123) [inline CONTRIBUTING.md]: No merge, only rebase

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+And any time after that:
+
+```bash
+git push
+```
+
+7. [Open a Pull Request](https://help.github.com/articles/using-pull-requests/) with a clear title and description against the `master` branch.
+
+**IMPORTANT**: By submitting a patch, you agree to allow the project owners to license your work under the terms of the [GPL v2](https://github.com/OpenTTD/OpenTTD/blob/master/COPYING).
```
Comment (TrueBrain) [inline CONTRIBUTING.md]: I really like the License part lower; this sentence feels a bit weird. Possibly just tell them to read the license part? :D

**Review by TrueBrain — CHANGES_REQUESTED** [review]

**Review by frosch123 — APPROVED** [review]

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**